### DISCUSSION TITLE
HttpClient

### DISCUSSION THREAD

**Discussion by msasanmh** [jchristn/RestWrapper#10] [Q&A] [answered]
Hi, I have a question in mind, do you have any plan to use HttpClient instead of HttpWebRequest?
I'm talking about this line:
https://github.com/jchristn/RestWrapper/blob/ac19d99cdf2885faaa1e83c73d5dce607a493307/src/RestWrapper/RestRequest.cs#L426

**Comment by jchristn** [comment]
Hi, I would like to, but I haven't found the time yet to migrate it.  If it's something you'd be interested in tackling I'd love to take a PR on it!

**Comment by msasanmh** [comment]
> Hi, I would like to, but I haven't found the time yet to migrate it.  If it's something you'd be interested in tackling I'd love to take a PR on it!

Actually I've tried and it kinda worked, but I had some issues.
It couldn't satisfy me, I deleted all the modifications.
I really don't want to mess it up. :(

**Reply by jchristn** [reply, marked as answer]
Hi @msasanmh I just completed the migration from ```HttpWebRequest``` to ```HttpClient``` in v3.0.  I'd love to have you take a look and let me know if there are any issues!

Commit: https://github.com/jchristn/RestWrapper/commit/993a29fd63f6d14ad26e62efb7d0ff028c62757d
NuGet: https://www.nuget.org/packages/RestWrapper/3.0.0

**Reply by msasanmh** [reply]
Thank you 🙏 I'm gonna check it out and let you know.

**Reply by msasanmh** [reply]
@jchristn The last time you asked me to pull a pr, I used HttpClient but because of a website called http://neverssl.com I revert back to HttpWebRequest.

but now I tried other non ssl websites, all of them are ok, like:
http://info.cern.ch
http://www.grose.us

except http://neverssl.com, but HttpWebRequest can open even this website correctly.

Error:
"The byte stream was erroneous according to the character encoding that was declared. The character encoding declaration may be incorrect."
![Neverssl com](https://github.com/jchristn/RestWrapper/assets/1586825/2e9d40ff-07f0-446d-b8e9-19f48f1483b5)

Conclusion:
Everything's OK except neverssl.com
Thank you for updating RestRequest.

**Reply by jchristn** [reply]
Hi @msasanmh thanks!  Interesting, I used the ```Test``` project with ```neverssl.com``` and it worked:
```
Command [? for help]: get
URL: [http://127.0.0.1:8000/] http://neverssl.com
REST Response
  Headers
  | Date: Wed, 12 Jul 2023 16:43:31 GMT
  | Server: Apache/2.4.57,()
  | Upgrade: h2,h2c
  | Connection: Upgrade,close
  | ETag: "f79-5e28b29d38e93"
  | Accept-Ranges: bytes
  | Vary: Accept-Encoding
  Content Type       : text/html; charset=UTF-8
  Status Code        : 200
  Status Description : OK
  Content Length     : 3961
  Time
  | Start (UTC)      : 2023-07-12 16:43:32
  | End (UTC)        : 2023-07-12 16:43:32
  | Total            : 409.97ms
  Data               : [stream]

Content:
<html>
        <head>
                <title>NeverSSL - Connecting ... </title>
                <style>
                body {
                        font-family: Montserrat, helvetica, arial, sans-serif;
                        font-size: 16x;
                        color: #444444;
                        margin: 0;
                }
                h2 {
                        font-weight: 700;
```
Thanks for taking a look!  Cheers, Joel

**Reply by msasanmh** [reply]
Yeah, status code and content are ok, I was rendering it on browser using proxy server, I guess it's because this website doesn't have a meta tag for encoding, like: `<meta charset="utf-8">`.
it's nothing important anyway.
Thanks again.
